library/general/test/popup_test.rb

```diff
@@ -62,9 +63,40 @@
   describe ".Error" do
     before { allow(ui).to receive(:OpenDialog) }
 
-    it "shows a popup without escaping tags" do
-      expect(subject).to receive(:RichText).with("<h1>Title</h1>")
-      subject.Error("<h1>Title</h1>")
+    let(:switch_to_richtext) { true }
+    let(:line) { "<h1>Title</h1>\n" }
+    let(:limit) { subject.too_many_lines }
+
+    around do |example|
+      old_switch_to_richtext = subject.switch_to_richtext
+      subject.switch_to_richtext = switch_to_richtext
+      example.run
+      subject.switch_to_richtext = old_switch_to_richtext
+    end
+
+    context "when switching to richtext is not allowed" do
+      let(:switch_to_richtext) { false }
+
+      it "shows a popup without escaping tags" do
+        message = line * limit
+        expect(subject).to receive(:RichText).with(message)
+        subject.Error(message)
+      end
+    end
+
+    context "when switch to richtext is allowed" do
+      let(:switch_to_richtext) { true }
+
+      it "escapes the tags if message is too long" do
+        message = line * limit
+        expect(subject).to receive(:RichText).with(Yast::String.EscapeTags(message))
+        subject.Error(message)
+      end
+
+      it "does not escape the tags if message is not too long" do
```

Comment:
NP: Too many `not` makes it less readable, IMHO *"keeps the original text if the message is short"* is better.